app/assets/javascripts/tasks_form.coffee

```diff
@@ -21,3 +22,11 @@ initializeFileTypeSelection = ->
     editor = $(this).parents('.file-container').find('.editor')[0]
     changeEditorMode(editor, getModeByFileExtension($(this).val()))
   $('.file-name').keyup()
+
+initializeVisibilityWarning = ->
+  warning_message = $('#task_visibility_warning')
+  $('#task_access_level_private').on 'change', ->
+    if warning_message.data("external-collection-membership")
+      warning_message.removeClass('hidden')
+  $('#task_access_level_public').on 'change', ->
+      warning_message.addClass('hidden')
```

Comment:
It would be great if you could add an empty line here. This is usually desired, since appending the file will then show correctly in a Git diff.